### ISSUE TITLE
Unsolicited response received on idle HTTP channel starting with ...

### ISSUE THREAD

**Issue by philomory** [m-kraus/check_nsc_web#7]
At various times, on running check_nsc_web, it will output "Unsolicited response received on idle HTTP channel starting with ..." and then a bunch of hex-escaped data (mostly null bytes). Afterwards, there will be a normal response code. For example:

```
2018/04/09 21:43:28 Unsolicited response received on idle HTTP channel starting with "\x00\x00\x00\f\x00\x00\x00\x02\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xb8\x9a\xaa\x01\x00\x00\x00\x00\xb8\x9a\xaa\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00": err=<nil>
OK: All 1 service(s) are ok.
```

To me, this appears likely to be related to https://github.com/golang/go/issues/19895; apparently the behavior of Go in this situation is intended, it's just surprising and unhelpful in this case. They do present some suggestions in that thread for how to adjust logging/output to avoid the situation, but I'm not really a Golang guy so I've not attempted to implement it myself yet.

**Comment by philomory** [comment]
Sorry, I should have mentioned, this is check_nsc_web v0.4.3, with NSClient++ 0.5.2.35.

Related to https://github.com/mickem/nscp/issues/550

**Comment by m-kraus** [comment]
Thank your for bringing this to my attention, especially linking to mickem/nscp#550

**Comment by m-kraus** [comment]
Should be closed with  mickem/nscp#550
